logs/client.go

```diff
@@ -0,0 +1,200 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"math/rand/v2"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/solarwinds/swo-cli/version"
+)
+
+type ColorStrFunc func(format string, a ...interface{}) string
+
+var colors = []ColorStrFunc{
+	color.CyanString,
+	color.YellowString,
+	color.GreenString,
+	color.MagentaString,
+	color.RedString,
+}
+
+type Client struct {
+	opts       *Options
+	httpClient http.Client
+	output     *os.File
+}
+
+type Log struct {
+	Time     time.Time `json:"time"`
+	Message  string    `json:"message"`
+	Hostname string    `json:"hostname"`
+	Severity string    `json:"severity"`
+	Program  string    `json:"program"`
+}
+
+type PageInfo struct {
+	PrevPage string `json:"prevPage"`
+	NextPage string `json:"nextPage"`
+}
+
+type LogsData struct {
+	Logs     []Log `json:"logs"`
+	PageInfo `json:"pageInfo"`
+}
+
+func NewClient(opts *Options) (*Client, error) {
+	return &Client{
+		httpClient: *http.DefaultClient,
+		opts:       opts,
+		output:     os.Stdout,
+	}, nil
+}
+
+func (c *Client) prepareRequest(ctx context.Context) (*http.Request, error) {
+	logsEndpoint, err := url.JoinPath(c.opts.ApiUrl, "v1/logs")
+	if err != nil {
+		return nil, err
+	}
+
+	logsUrl, err := url.Parse(logsEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	params := url.Values{}
+	params.Add("pageSize", strconv.Itoa(c.opts.count))
```

Comment:
Rather than asking for a single page of data with a configurable size, the CLI should be using a standard page size and looping through the results. It should make multiple calls to the API until all of the results have been retrieved.

Reply:
done